src/odemis/util/test/util_test.py

```diff
@@ -396,54 +401,255 @@ def test_line_clipping(self):
+
+
+class ExpandRectTestCase(unittest.TestCase):
+
+    def test_expand_rect(self):
+        """ Test expanding a rectangle """
+        rect = (5, 3, 10, 7)
+        expanded_rect = util.expand_rect(util.normalize_rect(rect), 5)
+        expected_rect = (0, -2, 15, 12)
+        self.assertEqual(expanded_rect, expected_rect,  "Failed to get correct output for rectangle %s with "
+                                                        "expanded rectangle %s" % (expanded_rect, expected_rect))
+
+
+class FindPlotContentTestCase(unittest.TestCase):
+
+    def test_find_plot_content(self):
+        """ Test finding left and right most non-zero values"""
+        xd = (10, 15, 20, 25, 30)
+        yd = (0, 0, 1, -2, 0)
+        plot_content = util.find_plot_content(xd, yd)
+        self.assertEqual(plot_content, (20, 25), "Failed to get correct output while finding plot content "
+                                                 "for %s with values  %s %s" % (plot_content, xd, yd))
+
+
+class WrapToMpiPpiTestCase(unittest.TestCase):
+
+    def test_positive_pi(self):
+        """ Test value is smaller than +pi value """
+        converted_angle = util.wrap_to_mpi_ppi(4.65)
+        self.assertLessEqual(converted_angle, math.pi, "Failed to convert the angle to a value between -pi and +pi "
+                                                  "with the converted value: %s" % converted_angle)
+
+    def test_negative_pi(self):
+        """ Test value is bigger than -pi value """
+        converted_angle = util.wrap_to_mpi_ppi(-5.1)
+        self.assertGreaterEqual(converted_angle, -math.pi, "Failed to convert the angle to a value between -pi and +pi "
+                                                      "with the converted value: %s" % converted_angle)
+
+
+class RecursiveDictUpdateTestCase(unittest.TestCase):
+
+    def test_dict_update(self):
+        """ Test recursively updating first dict with the second dict """
+        dict1 = {
+          "firstname": "John",
+          "lastname": "Doe",
+          "birthdate": 1964
+        }
+
+        dict2 = {
+            "suffix": "van",
+            "birthdate": 1994
+        }
+
+        expected_dict = {
+            "firstname": "John",
+            "suffix": "van",
+            "lastname": "Doe",
+            "birthdate": 1994
+        }
+
+        new_dict = util.recursive_dict_update(dict1, dict2)
+        self.assertEqual(new_dict, expected_dict, "Failed to get correct output while updating the dictionary "
+                                                  " %s" % new_dict)
+
+
+class PairwiseTestCase(unittest.TestCase):
+
+    def test_pairwise(self):
+        """ Test pairing """
+        test = util.pairwise(("item 1", "item 2"))
+        test_list = list(test)[0]
+        self.assertEqual(test_list, ("item 1", "item 2"), "Failed to pair.")
+
+
+class InspectGetMembersTestCase(unittest.TestCase):
+
+    def test_inspect_get_members(self):
+        """ Test inspect get members """
```

Comment:
This one is a bit more complex. I would update this to:
`""" Test the builtin inspect.getmembers raises a TypeError for the dummy class, and that util.inspect_getmembers has the correct output."""`